### ISSUE TITLE
MRMS interpolation bug

### ISSUE THREAD

**Issue by JeffHamiltonNOAA** [NOAA-GSL/VxLegacyIngest#524]
Amanda discovered an error in how our MRMS observation analysis is being interpolated and masked on Jet. Erroneous data appears to be introduced in domain grid points near where the mask is being applied and is then amplified by the interpolation to coarser resolutions. Investigate and correct

**Comment by JeffHamiltonNOAA** [comment]
After a meeting with Amanda, Curtis, Steve, and VAB members on Monday (3/6), I investigated the interpolation libraries on both Jet and Hera.

First the good news. I have isolated the problem to a mismatched configuration in the ipolates library between the two machines when dealing with grid-mask interactions. There are two methods for determining proper value placement and Hera was using an older subroutine that completed this process in one step, whereas Jet was using the newer method that split the subroutine into two.  I ran a test to have Jet do it using the old subroutine like Hera and the results look promising.

I'm going to run the same test on Amanda's case in late December (December 24) as well to be certain. 

Now the bad news. If the late December results are identical as well, this naturally leads us to conclude that there is a bug in the two new subroutines inside the ipolates library and has been there for some time (none of our developers have ever modified either). That will be my next thing to track down.

**Comment by JeffHamiltonNOAA** [comment]
Looking into the code further, the interpolation configuration was a red herring. The was actually a logic difference in how the mask is applied when weighting the average for eventual interpolation of the values. Jet was erroneously including the mask values (!!!!) in the interpolation, causing the zero values to appear around the mask edges. I've corrected, but am running both methods in parallel for the time being to get a sense of the impact on our statistics (per Steve's request). I still plan on rerunning the December 24 case since that is what Amanda used in her investigation

**Comment by JeffHamiltonNOAA** [comment]
I'm allowing this to run another week in order to get a larger scope of the impact. I'll verify against operational HRRR and provide slides of the differences for further discussion. Depending on those results, we may need to backfill the archive on HPSS, but that will lead to other issues with case comparisons to real time runs verified against the erroneous analysis at the time.

**Comment by JeffHamiltonNOAA** [comment]
After checking again this morning for the magnitude of the impact, I'm rerunning the past 5 days due to a typo I introduced in our XML file on Friday. This shouldn't take more than a few hours and I'll have results for all interested parties.

In any case, I modified the real time interpolation to use the correct method and this will be used moving forward. I'll open another issue on how to handle the archive once the magnitude is better understood

**Comment by JeffHamiltonNOAA** [comment]
I have completed the comparison and put together a few slides on this effort:

https://docs.google.com/presentation/d/1Q7swLc45ngRpy7ox6x5yr_blhtDWiwtG4DbnWI7E6aM/edit?usp=sharing

This issue is now closed, with the possibility of a follow up one if the division determines that work needs to be done to fix the archive.
